ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -4293,7 +4293,7 @@ private void deleteOldPathForReplace(Path destPath, Path oldPath, HiveConf conf,
       // But not sure why we changed not to delete the oldPath in HIVE-8750 if it is
       // not the destf or its subdir?
       isOldPathUnderDestf = isSubDir(oldPath, destPath, oldFs, destFs, false);
-      if (isOldPathUnderDestf) {
+      if (isOldPathUnderDestf && oldFs.exists(oldPath)) {
         cleanUpOneDirectoryForReplace(oldPath, oldFs, pathFilter, conf, purge, isNeedRecycle);
```

Comment:
Well I see it is a backport but about the fix, having an exist check increased on RPC for every genuine case as well, instead shouldn't we have ignored the FNF in `` cleanUpOneDirectoryForReplace``?

Reply:
Your comment makes sense to me. I think that is a valid way to handle this as well. But since this is a backport, if we want to do this we should do it in a separate PR in master branch first to avoid git conflicts in future.

Reply:
@ayushtkn Will you be able to approve this PR since you have already looked into this? I can create a follow-up PR which incorporates you feedback on master branch if you want.